packages/core/admin/admin/src/pages/Auth/components/ResetPassword.tsx

```diff
@@ -1,73 +1,110 @@
+} from '../../../layouts/UnauthenticatedLayout';
 
-const PasswordInput = styled(TextInput)`
-  ::-ms-reveal {
-    display: none;
-  }
-`;
+import { FieldActionWrapper } from './FieldActionWrapper';
 
-const ForgotPassword = ({ onSubmit, schema }) => {
-  const [passwordShown, setPasswordShown] = useState(false);
-  const [confirmPasswordShown, setConfirmPasswordShown] = useState(false);
+const REGISTER_SCHEMA = yup.object().shape({
+  password: yup
+    .string()
+    .min(8, translatedErrors.minLength)
+    .matches(/[a-z]/, 'components.Input.error.contain.lowercase')
+    .matches(/[A-Z]/, 'components.Input.error.contain.uppercase')
+    .matches(/\d/, 'components.Input.error.contain.number')
+    .required(translatedErrors.required),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref('password'), null], 'components.Input.error.password.noMatch')
+    .required(translatedErrors.required),
+});
+
+const ResetPassword = () => {
+  const [passwordShown, setPasswordShown] = React.useState(false);
+  const [confirmPasswordShown, setConfirmPasswordShown] = React.useState(false);
   const { formatMessage } = useIntl();
+  const { post } = useFetchClient();
+  const { push } = useHistory();
+  const query = useQuery();
+
+  const { mutate, isError } = useMutation(
+    async (body: Pick<ResetPassword.Request['body'], 'password'>) => {
+      const {
+        data: { data },
+      } = await post<ResetPassword.Response>('/admin/reset-password', {
+        ...body,
+        resetPasswordToken: query.get('code'),
+      });
+
+      return data;
+    },
+    {
+      onSuccess(data) {
+        if (data) {
+          const { token, user } = data;
+          auth.setToken(token, false);
+          auth.setUserInfo(user, false);
+
+          // Redirect to the homePage
+          push('/');
+        }
+      },
+    }
+  );
 
   return (
     <UnauthenticatedLayout>
       <Main>
         <LayoutContent>
+          <Column>
+            <Logo />
+            <Box paddingTop={6} paddingBottom={7}>
+              <Typography as="h1" variant="alpha">
+                {formatMessage({
+                  id: 'global.reset-password',
+                  defaultMessage: 'Reset password',
+                })}
+              </Typography>
+            </Box>
+            {isError ? (
+              <Typography id="global-form-error" role="alert" tabIndex={-1} textColor="danger600">
+                {formatMessage({
+                  id: 'notification.error',
+                  defaultMessage: 'An error occurred',
+                })}
+              </Typography>
+            ) : null}
+          </Column>
           <Formik
             enableReinitialize
             initialValues={{
               password: '',
               confirmPassword: '',
             }}
-            onSubmit={onSubmit}
-            validationSchema={schema}
+            onSubmit={(values) => {
+              mutate(values);
```

Comment:
```suggestion
              mutate({ password: values.password });
```

Otherwise the BE endpoint is failing as it receives 'confirmPassword' too